ovs/extensions/healthcheck/openvstorage/openvstoragecluster_health_check.py

```diff
@@ -170,91 +162,92 @@ def _list_dirs_in_directory(pwd):
-            pcel = commands.getoutput(pcommand.format(pcommand)).splitlines()
-            if len(pcel) != 1 and 'pong' in pcel[1].strip():
-                logger.success('Connection successfully established!', 'port_celery')
+        if Helper.get_ovs_type() != 'MASTER':
+            result_handler.skip('RabbitMQ is not running/active on this server!', 'port_celery')
+        result_handler.info('Checking Celery.')
+        from errno import errorcode
+        try:
+            from celery.task.control import inspect
+            stats = inspect().stats()
+            if stats:
+                result_handler.success('Successfully connected to Celery on all nodes.', test_name)
             else:
-                logger.failure('Connection FAILED to service Celery, please check RabbitMQ and ovs-workers?',
-                               'port_celery')
-        else:
-            logger.skip('RabbitMQ is not running/active on this server!', 'port_celery')
+                result_handler.failure('No running Celery workers were found.', test_name)
+        except IOError as ex:
+            msg = 'Could not connect to Celery. Got {0}.'.format(ex)
+            if len(ex.args) > 0 and errorcode.get(ex.args[0]) == 'ECONNREFUSED':
+                msg += ' Check that the RabbitMQ server is running.'
+                result_handler.failure(msg, test_name)
+        except ImportError as ex:
+            result_handler.failure('Could not import the celery module. Got {}'.format(str(ex)), test_name)
 
     @staticmethod
     @expose_to_cli('ovs', 'packages-test')
-    def check_ovs_packages(logger):
+    def check_ovs_packages(result_handler):
         """
         Checks the availability of packages for Open vStorage
 
-        :param logger: logging object
-        :type logger: ovs.log.healthcheck_logHandler.HCLogHandler
+        :param result_handler: logging object
+        :type result_handler: ovs.extensions.healthcheck.result.HCResults
         """
 
-        logger.info('Checking OVS packages: ', 'check_ovs_packages')
+        result_handler.info('Checking OVS packages: ', 'check_ovs_packages')
 
         for package in Helper.packages:
-            result = commands.getoutput('apt-cache policy {0}'.format(package)).splitlines()
-            if len(result) != 1:
-                logger.success(
-                    'Package {0} is present, with version {1}'.format(package, result[2].split(':')[1].strip()),
-                    'package_{0}'.format(package))
+            cmd = ['apt-cache', 'policy', package]
+            process = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            out, err = process.communicate()
+            results = out.splitlines()
+            if len(results) != 1:
+                result_handler.success('Package {0} is present, with version {1}'.format(package, results[2].split(':')[1].strip()),
```

Comment:
The version printed is the candidate version not the actually installed version
In order to retrieve the installed version it might be better to use PackageManager.get_installed_versions()